### PR TITLE
build: add most components to kitchen-sink

### PR DESCRIPTION
**Pull request by jelbourn**
Fixes a handful of issues that prevent some components from working.

Checkbox and button-toggles don't work due to [an issue in platform-server](https://github.com/angular/angular/issues/17050).

Tabs don't work because they try to set a `style` property directly on initialization; this will need a more in-depth follow-up (I didn't look into that deeply). 

Related to #308

### CODE REVIEW

File: src/lib/core/a11y/interactivity-checker.ts [hunk 1]
```diff
@@ -48,6 +48,10 @@ export class InteractivityChecker {
    * @returns Whether the element is tabbable.
    */
   isTabbable(element: HTMLElement): boolean {
+    // Nothing is tabbable on the the server 😎
```
Comment (DevVersion) [inline src/lib/core/a11y/interactivity-checker.ts]: We need a lint rule for that! 😄

File: src/lib/slide-toggle/slide-toggle.ts [hunk 2]
```diff
@@ -312,9 +314,13 @@ class SlideToggleRenderer {
   /** Whether the thumb is currently being dragged. */
   dragging: boolean = false;
 
-  constructor(private _elementRef: ElementRef) {
-    this._thumbEl = _elementRef.nativeElement.querySelector('.mat-slide-toggle-thumb-container');
-    this._thumbBarEl = _elementRef.nativeElement.querySelector('.mat-slide-toggle-bar');
+  constructor(private _elementRef: ElementRef, platform: Platform) {
+    // We only need to interact with these elements when we're on the browser, so only grab
+    // the refernece in that case.
```
Comment (DevVersion) [inline src/lib/slide-toggle/slide-toggle.ts]: Typo for `reference`.

> Also we are already in the custom renderer? I suppose the whole custom renderer approach will be addressed in the future at some point.

> Just thought that we could not create the Renderer if we are not in the browser

Comment (jelbourn) [inline src/lib/slide-toggle/slide-toggle.ts]: That's the long-term plan, but that will require the secondary `@angular/material/server` package

Comment (DevVersion) [inline src/lib/slide-toggle/slide-toggle.ts]: Okay sounds good

**Comment by jelbourn** [comment]
Still trying to fix these test failures, which happen inconsistently locally

**Comment by jelbourn** [comment]
All failures resolved

**Pull request merged**